flexget/components/notify/notifiers/matrix.py

```diff
@@ -0,0 +1,80 @@
+import logging
+
+from loguru import logger
+
+from flexget import plugin
+from flexget.config_schema import one_or_more
+from flexget.event import event
+from flexget.plugin import DependencyError, PluginWarning
+
+
+plugin_name = 'matrix'
+
+logger = logger.bind(name=plugin_name)
+
+
+class MatrixNotifier:
+    """
+    Sends messages via Matrix. The MatrixHttpApi library is required to be installed.
+    Install it with: `pip install matrix_client`
+
+    All fields are required.
+
+    token: Is available in Element under Help/About.
+    roomId: Is available in Element under room settings.
+
+    Example::
+
+      notify:
+        entries:
+          via:
+            - matrix:
+                server: "https://matrix.org"
+                token: sender's token
+                roomid: room identifier
+    """
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'server': {'type': 'string'},
+            'token': {'type': 'string'},
+            'roomid': {'type': 'string'},
+        },
+        'required': ['server', 'token', 'roomid'],
+        'additionalProperties': False,
+    }
+
+    __version__ = '0.1'
+
+    def notify(self, title, message, config):
+        try:
+            from matrix_client.api import MatrixHttpApi  # noqa
+        except ImportError as e:
+            logger.debug('Error importing MatrixHttpApi: {}', e)
+            raise DependencyError(
+                plugin_name, 'matrix_client', 'matrix_client module required. ImportError: %s' % e
+            )
+        #TODO: Is dns required to be loaded for the matrix_client module?
+        try:
+            import dns  # noqa
```

Comment:
Don't believe this import of DNS is needed. As it's never implemented into the call, and all DNS Queries would be handled by the MatrixHttpApi module and not by flexget.

Reply:
I've updated to remove the external deps